doc/guides/build/index.rst

```diff
@@ -0,0 +1,134 @@
+.. _build_overview:
+
+Build Overview
+##############
+
+The Zephyr build process can be divided into two main phases: a
+configuration phase (driven by *CMake*) and a build phase (driven by
+*Make* or *Ninja*). This document is describing the build using *Make*
+as example.
+
+Configuration Phase
+*******************
+
+The configuration phase begins when the user invokes *CMake*,
+specifying a source application directory and a board target.
+
+.. figure:: build-config-phase.png
+    :align: center
+    :alt: Zephyr's build configuration phase
+    :figclass: align-center
+    :width: 80%
+
+*CMake* begins by processing the *CMakeLists.txt* file in the application
+directory, which will refer to the *CMakeLists.txt* file in the Zephyr
+top-level directory, which in turn will refer to *CMakeLists.txt* files
+throughout the build tree (directly and indirectly). Its primary
+output is a set of *Makefile*  to drive the build process, but *CMake*
+scripts do some build processing of their own:
+
+**Device tree** Using *cpp*, device-tree specifications (.dts files)
+are collected from the target’s architecture, SoC, board, and
+application directories and compiled with *dtc*. Then the build tool
+(scripts/dts) convert this into .conf files and .h files for later
+consumption.
```

Comment:
> and sets numerous DT_* that alter CMake's ultimate Makefile output.

Adding this would show that there is a pattern between DT and Kconfig.
But it would also describe a feature that is not much used, and that some are even
trying to remove (https://github.com/zephyrproject-rtos/zephyr/pull/20318/files#r342450803).

So it can be omitted if brevity is more important than being comprehensive.

Reply:
just removed, thanks for pointing it out.